File: backend/src/game/objects/brick.rs
use crate::game::{config::BRICK_RUIN_COUNTDOWN_MS, types::{MapIndex, ObjType}};
use super::base::BaseObj;
use crate::game::objects::base::BaseObjTrait;
pub struct BrickObj {
    pub base: BaseObj,
    pub is_ruining: bool,
    pub remaining_ms:u32,
}

impl BrickObj {
    pub fn new(index: &MapIndex) -> Self {
        Self {
            base: BaseObj::new(index, ObjType::Brick),
            is_ruining: false,
            remaining_ms:BRICK_RUIN_COUNTDOWN_MS
        }
    }

    pub fn trigger_ruin(&mut self) -> MapIndex {
        self.is_ruining = true;
        self.base.get_map_index()
    }
}

File: py-game/game/objects/brick.py
from config import COUNTDOWN_BRICK_RUIN_MS
from game.objects.base import BaseObj


class BrickObj(BaseObj):
    def __init__(self, index_x: int, index_y: int):
        super().__init__(index_x, index_y, "brick")
        self.is_ruining = False
        self.ruin_timer_ms = 0

    def trigger_ruin(self):
        self.is_ruining = True
        self.ruin_timer_ms = COUNTDOWN_BRICK_RUIN_MS
